packages/block-library/src/navigation/index.php

```diff
@@ -190,11 +190,19 @@ private static function get_inner_blocks_from_navigation_post( $attributes ) {
 
 			// 'parse_blocks' includes a null block with '\n\n' as the content when
 			// it encounters whitespace. This code strips it.
-			$compacted_blocks = block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
+			$blocks = block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
+
+			if ( function_exists( 'get_hooked_blocks' ) ) {
+				// Run Block Hooks algorithm to inject hooked blocks.
+				$markup         = gutenberg_insert_hooked_blocks_into_navigation_block( $blocks, $navigation_post );
```

Comment:
We'll need to change this function name: Dynamic blocks PHP -- i.e. pretty much any `packages/block-library/src/*/*.php` files -- cannot contain any `gutenberg_` prefixed functions, as they are synced verbatim from GB to Core via npm package sync.

Reply:
https://github.com/WordPress/gutenberg/pull/57754/commits/2211001ef22f8369d9422426e91b27442bf73625